game/assets/scripts/dialogue/dialogue_player.lua

```diff
@@ -1,76 +1,170 @@
 
-exports = {
-	story = ""
-}
+function DialoguePlayer:initialize(entity)
+    self.exports = {
+		story = "",
+		up = "Up",
+		down = "Down",
+		next = "Next"
+	}
+end
+
+function DialoguePlayer:begin(entity)
+	self.narrator = require("narrator/narrator")
+	local book = self.narrator.parseFile(self.exports.story)
+	self.story = self.narrator.initStory(book)
+	self.story:begin()
+	self.ui = entity:getUI()
+	self.currentChoice = 1
+	self.maxChoices = 1
+	self.currentActor = ""
+	if self.story.variables["default_color"] then
+		self.default_color = self.story.variables["default_color"]
+	else 
+		self.default_color = "white"
+	end
+	if self.story.variables["default_actor_color"] then
+		self.default_actor_color = self.story.variables["default_actor_color"]
+	else 
+		self.default_actor_color = "white"
+	end
+	self.actors = {
+		["smiley"] = {
+			["happy"] = "images/smiley_happy.jpg",
+			angry = "images/smiley_angry.jpg",
+			sad = "images/smiley_sad.jpg"
+		}
+	}
+	self:getDocument():Show()
+end
+
+function DialoguePlayer:update(entity, delta)
+	if self.story:canContinue() then
+		local paragraphs = self.story:continue()
+		self:presentParagraphs(paragraphs)
+		if self.story:canChoose() then
+			self:presentChoices(self.story:getChoices())
+		end
+	end
 
-function onBegin(entity)
-	local book = narrator.parseFile(exports.story)
-	story = narrator.initStory(book)
-	story:begin()
-	ui = entity:getUI()
-	getDocument():Show()
+	if self.story:canChoose() then
+		if RTX.Input.WasPressed(self.exports.up) then
+			self.currentChoice = self.currentChoice - 1
+			if self.currentChoice < 1 then self.currentChoice = self.maxChoices end
+			self:presentChoices(self.story:getChoices())
+		end
+		
+		if RTX.Input.WasPressed(self.exports.down) then
+			self.currentChoice = self.currentChoice + 1
+			if self.currentChoice > self.maxChoices then self.currentChoice = 1 end
+			self:presentChoices(self.story:getChoices())
+		end
+		
+		if RTX.Input.WasPressed(self.exports.next) then
+			local choices = self.story:getChoices()
+			if choices[self.currentChoice+1].tags then
+				for _, tag in ipairs(choices[self.currentChoice].tags) do
+					local args = {}
+					for str in string.gmatch(tag, "[^%s]+") do
+							table.insert(args, str)
+					end
+					if args[1] == "onselect" then
+						RTX.CallEvent(RTX.Event.new(args[2], args[3]))
+						print("event " .. args[2] .. " emitted")
+					end
+				end
+			end
+			self.story:choose(self.currentChoice)
+			self.currentChoice = 1
+		end
+	else
+		self:getDocument():Hide()
+	end
 end
 
-function getDocument()
-	return rmlui.contexts["default"].documents[ui:getDocumentID()]
+
+function DialoguePlayer:getDocument()
+	return rmlui.contexts["default"].documents[self.ui:getDocumentID()]
 end
 
-function presentParagraphs(paragraphs)
+function DialoguePlayer:presentParagraphs(paragraphs)
 	local text = ""
+	local actor = ""
+	local color = self.default_color
+	local actorColor = self.default_actor_color
+	local class = ""
```

Comment:
Avoid the name `class` because `class` as a global variable already exists (see the topmost line). 

This is not an error because of the `local` keyword but someone can see this and create a `class` variable without declaring it local and suddenly now we can't make classes in Lua.